### PR TITLE
Fix: func_healthcharger, func_recharge, func_break, func_tank

### PR DESCRIPTION
**Pull request by Vaqtincha**

### CODE REVIEW

File: regamedll/dlls/mapinfo.cpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ void CMapInfo::KeyValue(KeyValueData *pkvd)
 	}
 	else if (FStrEq(pkvd->szKeyName, "bombradius"))
 	{
-		m_flBombRadius = (float)Q_atoi(pkvd->szValue);
+		m_flBombRadius = Q_atof(pkvd->szValue);
```
Comment (s1lentq) [inline regamedll/dlls/mapinfo.cpp]: Behavior changed, float rounded here, so you have 2 way
1. Leave this
2. Wrap in a macro REGAMEDLL_FIXES

File: regamedll/dlls/h_battery.cpp [hunk 2]
```diff
@@ -67,11 +67,6 @@ void CRecharge::Restart()
 	UTIL_SetSize(pev, pev->mins, pev->maxs);
 	SET_MODEL(ENT(pev), STRING(pev->model));
 
-	int armorValue = (int)gSkillData.suitchargerCapacity;
-	if (pev->armorvalue != 0.0f) {
-		armorValue = (int)pev->armorvalue;
-	}
-
```
Comment (s1lentq) [inline regamedll/dlls/h_battery.cpp]: Why was this removed?

File: regamedll/dlls/maprules.cpp [hunk 3]
```diff
@@ -185,7 +185,7 @@ void CGameText::Spawn()
 {
 #ifdef REGAMEDLL_FIXES
 	// Don't allow entity triggering itself
-	if (FStrEq(pev->target, pev->targetname))
```
Comment (s1lentq) [inline regamedll/dlls/maprules.cpp]: This change is pointless, because `pev->target` or `pev->targetname` are `QString` class that those have operator `operator const char *()`

File: regamedll/dlls/h_battery.cpp (hunk 2 above)
Comment (Vaqtincha) [reply regamedll/dlls/h_battery.cpp]: Not used. 

Not necessary. (Func "Recharge")

File: regamedll/dlls/maprules.cpp (hunk 3 above)
Comment (Vaqtincha) [reply regamedll/dlls/maprules.cpp]: Ok.

**Comment by Vaqtincha** [comment]
:cry:
